src/iframe/storage/unpartitioned-storage/local-storage.ts

```diff
@@ -42,4 +89,536 @@ export class LocalStorage extends EnhancedStorage {
 
     return this.instance;
   }
+
+  /**
+   * Determine the primary storage strategy based on access state
+   * @returns Whether to use cookies as primary storage
+   */
+  /*
+  private shouldUseCookiesAsPrimary(): boolean {
+    // Only use cookies as primary when we have cookie access but not full access
+    return this.storage.hasCookieAccess() && !this.storage.hasFullAccess();
+  }
+  */
+
+  /**
+   * Returns `true` if the given key should be stored using cookies instead of localStorage. This only happens for the
+   * device nonce and the Supabsae auth token if and only if the unpartitioned state status is "limited", meaning,
+   * unpartitioned state is supported but only for cookies.
+   *
+   * In any other case, like not having unpartitioned state support at all or if unpartitioned state for localStorage
+   * is supported, we do not use cookies.
+   */
+  private shouldStoreInCookies(key: string): boolean {
+    if (getUnpartitionedStateStatus() !== "limited") return false;
+
+    return key === DEVICE_NONCE_KEY || SUPABASE_AUTH_TOKEN_KEY_REGEXP.test(key);
+  }
+
+  /**
+   * Get access to the CookieStore API safely
+   */
+  private getCookieStore() {
+    if (this.cookieStoreAvailable) return window.cookieStore;
+    return undefined;
+  }
+
+  /**
+   * Set a cookie using CookieStore API or js-cookie fallback
+   */
+  private async setCookie(key: string, value: string, options?: CookieAttributes): Promise<boolean> {
+    const cookieOptions = options || this.COOKIE_OPTIONS;
+
+    try {
+      const cookieStore = this.getCookieStore();
+      if (cookieStore) {
+        const cookieOptions = {
+          name: key,
+          value,
+          ...this.COOKIE_STORE_OPTIONS,
+        };
+
+        await cookieStore.set(cookieOptions);
+        const result = await this.getCookieValue(key);
+        if (result !== value) throw new Error(`Failed to set cookie for key '${key}'`);
+        return result === value;
+      } else {
+        // Fallback to js-cookie
+        Cookies.set(key, value, cookieOptions);
+        const result = Cookies.get(key);
+        return result === value;
+      }
+    } catch (e) {
+      if (LocalStorage.DEBUG_MODE) console.warn(`Failed to set cookie for key '${key}'`, e);
+
+      /*
+      try {
```

Comment:
Is there any reason for so many fallbacks? There's already the `if (cookieStore)` check above, but we still try to use `Cookies` in the `catch` block, and then also try with `localStorage`. Same question applies to the other methods that have this same pattern.

Reply:
I think I used cookieStore but it could throw error so added a fallback with `js-cookie` incase if it failed in the catch block. I think regarding localStorage, i was trying to sync the localStorage and cookies so they have the same value if in a case where the storage can switch from cookie to localStorage if permissions was granted later.